455 conclusion

ChenghengLi opened the pull request:
<!--
    Your PR title can should describe what feature was added/changed
-->

## Summary

Closes #455 

Page conclusion created
<!-- Enumerate changes you made and why you made them in bullet form-->
<!-- list any new dependencies required for this change -->

## Screenshots

<img width="524" alt="image" src="https://github.com/uclaacm/parcel-pointers/assets/80062217/8815b2a9-888f-4d66-bc6d-05af85e4dd07">

<img width="1440" alt="image" src="https://github.com/uclaacm/parcel-pointers/assets/80062217/1a378276-c9b2-4f44-831f-0e00504887b7">

<!--
    Add Screenshots of the feature in play, terminal pastes, etc. as necessary
-->

ChenghengLi committed:
Innitial commit

ChenghengLi committed:
finish component

ChenghengLi committed:
fix lint

Dankco commented on src/pages/Conclusion.tsx:
```diff
@@ -0,0 +1,45 @@
+          </div>
+        </div>
+      </AppWrapper>
+      <NavButtons page={16}></NavButtons>
```
Going to split this out as a different ticket, but we should not have a next arrow for this page since it is the last one.

Dankco reviewed (commented):
Can you add the page to the sidebar here after the exercises?
https://github.com/uclaacm/parcel-pointers/blob/main/src/components/Sidebar.tsx#L165

ChenghengLi committed:
remove button

ChenghengLi commented:
Just fixed the button issue, also removed the prev button for the first exercise and lesson; the next button for the last lesson.

Also added the conclusion section to the menu.

NOTE: Added two props to NavButtons"next" and "prev" as booleans to indicate the visibility of the buttons.

ChenghengLi committed:
fix index

Dankco commented on src/components/Sidebar.tsx:
```diff
@@ -113,6 +113,10 @@ export default function Sidebar(props: sidebarProps): JSX.Element {
       name: '3. Dereferencing Pointers',
       link: PageURLs.EXERCISE_3,
     },
+    {
```
Can you move this outside of the list?
The conclusion is not an exercise, so it shouldn't go in the exercises dropdown.
<img width="352" alt="image" src="https://github.com/uclaacm/parcel-pointers/assets/78634083/2f24d58b-c7d8-4fb1-8e62-e12379f3f8ab">

Dankco commented on src/components/NavButtons.tsx:
```diff
@@ -4,20 +4,24 @@ import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
 export interface NavProps {
   page: number;
+  next: boolean;
+  prev: boolean;
```
Instead of making these mandatory, you can make them optional and either set a default value or make them appear unless next/prev are set to false so that we don't need them in every page.

Dankco commented:
Also, let's keep the prev/next buttons for the exercises and lesson so there is a single "flow" through all the pages that goes home -> motivation -> demo -> lessons -> exercises -> conclusion, so we would only need to not have a prev button at the home (which there isn't since there are no nav buttons, and no next button at the conclusion, which you just did.

Dankco commented:
Actually, I will take over and make the fixes here. You can start on the lesson 4 page.

Dankco committed:
change navbuttons to default to not needing props and move conclusion in sidebar

Dankco committed:
fix lint

Dankco reviewed (approved):
Nice work here, it looks good!

Dankco committed:
Merge branch 'main' into 455_Conclusion
